### PR TITLE
KASM Containers

### PR DESCRIPTION
**Pull request by justinglock40**

### CODE REVIEW

File: apps/kasmworkspace/chrome.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    hostname: 'chrome'
+    image: 'kasmweb/chrome:1.9.0-rolling'
+    restart: '${RESTARTAPP}'
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
```
Comment (doob187) [inline apps/kasmworkspace/chrome.yml]: No Hardcoded Value

File: apps/kasmworkspace/discord.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    hostname: 'discord'
+    image: 'kasmweb/discord:1.9.0-rolling'
+    restart: '${RESTARTAPP}'
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
```
Comment (doob187) [inline apps/kasmworkspace/discord.yml]: Same.here

File: apps/kasmworkspace/discord.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6905:6901'
```
Comment (doob187) [inline apps/kasmworkspace/discord.yml]: BLOCKPORT missed

File: apps/kasmworkspace/chrome.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6902:6901'
```
Comment (doob187) [inline apps/kasmworkspace/chrome.yml]: BLOCKPORT missed

File: apps/kasmworkspace/discord.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6905:6901'  
+    security_opt:
+      - 'no-new-privileges:true'
+    volumes:
+      - '/etc/localtime:/etc/localtime:ro'
+      - '${APPFOLDER}/discord:/config:rw'
+      - '/mnt:/mnt'
```
Comment (doob187) [inline apps/kasmworkspace/discord.yml]: Use unionfs Local persistent mode

File: apps/kasmworkspace/firefox.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
+    networks:
+      - proxy
+    ports:
+      - '6906:6901'
```
Comment (doob187) [inline apps/kasmworkspace/firefox.yml]: BLOCKPORT again

File: apps/kasmworkspace/firefox.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6906:6901'  
+    security_opt:
+      - 'no-new-privileges:true'
+    volumes:
+      - '/etc/localtime:/etc/localtime:ro'
+      - '${APPFOLDER}/firefox:/config:rw'
+      - '/mnt:/mnt'
```
Comment (doob187) [inline apps/kasmworkspace/firefox.yml]: Same here use Local persistent

File: apps/kasmworkspace/onlyoffice.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    hostname: 'onlyoffice'
+    image: 'kasmweb/onlyoffice:1.9.0-rolling'
+    restart: '${RESTARTAPP}'
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
```
Comment (doob187) [inline apps/kasmworkspace/onlyoffice.yml]: Password need value in env

File: apps/kasmworkspace/onlyoffice.yml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6909:6901'
```
Comment (doob187) [inline apps/kasmworkspace/onlyoffice.yml]: BLOCKPORT  again

File: apps/kasmworkspace/signal.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    hostname: 'signal'
+    image: 'kasmweb/signal:1.9.0-rolling'
+    restart: '${RESTARTAPP}'
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
```
Comment (doob187) [inline apps/kasmworkspace/signal.yml]: Value again

File: apps/kasmworkspace/signal.yml [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6904:6901'
```
Comment (doob187) [inline apps/kasmworkspace/signal.yml]: BLOCKPORT again

File: apps/kasmworkspace/steam.yml [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    image: 'kasmweb/steam:1.9.0-rolling'
+    restart: '${RESTARTAPP}'
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
+    networks:
```
Comment (doob187) [inline apps/kasmworkspace/steam.yml:16]: Env again

File: apps/kasmworkspace/steam.yml [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+      - proxy
+    ports:
+      - '6903:6901'
```
Comment (doob187) [inline apps/kasmworkspace/steam.yml]: BLOCKPORT

File: apps/kasmworkspace/telegram.yml [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+    hostname: 'telegram'
+    image: 'kasmweb/telegram:1.9.0-rolling'
+    restart: '${RESTARTAPP}'
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
```
Comment (doob187) [inline apps/kasmworkspace/telegram.yml]: Emv

File: apps/kasmworkspace/telegram.yml [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6905:6901'  
+    security_opt:
+      - 'no-new-privileges:true'
+    volumes:
+      - '/etc/localtime:/etc/localtime:ro'
+      - '${APPFOLDER}/telegram:/config:rw'
+      - '/mnt:/mnt'
```
Comment (doob187) [inline apps/kasmworkspace/telegram.yml]: Local persistent

File: apps/kasmworkspace/telegram.yml [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
+    networks:
+      - proxy
+    ports:
+      - '6905:6901'
```
Comment (doob187) [inline apps/kasmworkspace/telegram.yml]: BLOCKPORT

File: apps/kasmworkspace/tor.yml [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+    hostname: 'tor'
+    image: 'kasmweb/tor:1.9.0-rolling'
+    restart: '${RESTARTAPP}'
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
```
Comment (doob187) [inline apps/kasmworkspace/tor.yml]: Env

File: apps/kasmworkspace/tor.yml [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6908:6901'
```
Comment (doob187) [inline apps/kasmworkspace/tor.yml]: BLOCKPORT

File: apps/kasmworkspace/tor.yml [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6908:6901'  
+    security_opt:
+      - 'no-new-privileges:true'
+    volumes:
+      - '/etc/localtime:/etc/localtime:ro'
+      - '${APPFOLDER}/tor:/config:rw'
+      - '/mnt:/mnt'
```
Comment (doob187) [inline apps/kasmworkspace/tor.yml]: Local persistent

File: apps/kasmworkspace/vlc.yml [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+    hostname: 'vlc'
+    image: 'kasmweb/vlc:1.9.0-rolling'
+    restart: '${RESTARTAPP}'
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'VNC_RESOLUTION=1920x1080'
+      - 'VNC_PW=password'
```
Comment (doob187) [inline apps/kasmworkspace/vlc.yml]: Env

File: apps/kasmworkspace/vlc.yml [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6907:6901'
```
Comment (doob187) [inline apps/kasmworkspace/vlc.yml]: BLOCKPORT

File: apps/kasmworkspace/vlc.yml [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      - proxy
+    ports:
+      - '6907:6901'  
+    security_opt:
+      - 'no-new-privileges:true'
+    volumes:
+      - '/etc/localtime:/etc/localtime:ro'
+      - '${APPFOLDER}/vlc:/config:rw'
+      - '/mnt:/mnt'
```
Comment (doob187) [inline apps/kasmworkspace/vlc.yml]: Local persistent

**Review by doob187 — CHANGES_REQUESTED** [review]
Fix PLZ

**Review by doob187 — APPROVED** [review]
👍 from.my side

@drag0n141

**Review by drag0n141 — APPROVED** [review]
Looks good :)

**Pull request merged**